src/modules/vendor/components/programs/cards/templates/Survey.vue

```diff
@@ -1,24 +1,40 @@
+  <div class="container">
     <ni-input caption="Question" v-model="card.question" required-field @focus="saveTmp('question')"
       @blur="updateCard('question')" :error="v$.card.question.$error" :error-message="questionErrorMsg"
       :disable="disableEdition" />
-    <q-checkbox v-model="card.isMandatory" @update:model-value="updateCard('isMandatory')" label="Réponse obligatoire"
-      class="q-mb-lg" dense :disable="disableEdition" />
-    <div class="row gutter-profile">
-      <ni-input class="col-md-6 col-xs-12" caption="Légende niveau 1" v-model="card.labels['1']"
-        @focus="saveTmp('labels.1')" @blur="updateCardLabels('1')" :error="v$.card.labels['1'].$error"
-        :error-message="labelErrorMessage('1')" :disable="disableEdition" required-field />
-      <ni-input class="col-md-6 col-xs-12" caption="Légende niveau 5" v-model="card.labels['5']"
-        @focus="saveTmp('labels.5')" @blur="updateCardLabels('5')" :error="v$.card.labels['5'].$error"
-        :error-message="labelErrorMessage('5')" :disable="disableEdition" required-field />
+    <div class="checkbox-container">
+      <q-checkbox v-model="card.isMandatory" @update:model-value="updateCard('isMandatory')" label="Réponse obligatoire"
+        class="q-mb-lg" dense :disable="disableEdition" />
+      <q-checkbox :model-value="displayAllLabels" @update:model-value="initializeCardLabels"
+        label="Définir les légendes de chaque niveau" class="q-mb-lg" dense :disable="disableEdition" />
+    </div>
+    <div class="input-container">
+      <ni-input class="input" caption="Légende niveau 1" v-model="card.labels['1']" @focus="saveTmp('labels.1')"
+        @blur="updateCardLabels('1')" :error="v$.card.labels['1'].$error" :error-message="labelErrorMessage('1')"
+        :disable="disableEdition" required-field />
+      <template v-if="displayAllLabels">
+        <ni-input caption="Légende niveau 2" v-model="card.labels['2']"
+          @focus="saveTmp('labels.2')" @blur="updateCardLabels('2')" :error="v$.card.labels['2'].$error"
+          :error-message="labelErrorMessage('2')" :disable="disableEdition" required-field />
+        <ni-input caption="Légende niveau 3" v-model="card.labels['3']"
+          @focus="saveTmp('labels.3')" @blur="updateCardLabels('3')" :error="v$.card.labels['3'].$error"
+          :error-message="labelErrorMessage('3')" :disable="disableEdition" required-field />
+        <ni-input v-if="displayAllLabels" caption="Légende niveau 4" v-model="card.labels['4']"
+          @focus="saveTmp('labels.4')" @blur="updateCardLabels('4')" :error="v$.card.labels['4'].$error"
+          :error-message="labelErrorMessage('4')" :disable="disableEdition" required-field />
+      </template>
+      <ni-input caption="Légende niveau 5" v-model="card.labels['5']" @focus="saveTmp('labels.5')"
+        @blur="updateCardLabels('5')" :error="v$.card.labels['5'].$error" :error-message="labelErrorMessage('5')"
+        :disable="disableEdition" required-field />
```

Comment:
est ce qu'il n'y a pas une boucle for a faire ici ?

Reply:
Si bien vu !